EventsExpress.Core/Services/EventService.cs

```diff
@@ -347,6 +347,10 @@ public IEnumerable<EventDto> GetAll(EventFilterViewModel model, out int count)
                 ? events.Where(x => x.Visitors.Any(v => v.UserId == model.VisitorId))
                 : events;
 
+            events = (model.X != null && model.Y != null && model.Radius != null)
```

Comment:
Consider using the `Distance` method instead of calculating it manually. E.g.: https://www.c-sharpcorner.com/article/spatial-queries-in-entity-framework-core/